src/read_write/s2.rs

```diff
@@ -120,35 +131,37 @@ where
         }
         self.writers.get_mut(cell_id).unwrap()
     }
-}
 
-pub fn s2_cloud_to_meta_proto(
-    cells: Vec<proto::S2Cell>,
-    attributes: &BTreeMap<String, AttributeData>,
-) -> proto::Meta {
-    let mut meta = proto::Meta::new();
-    meta.set_version(CURRENT_VERSION);
-    let mut s2_meta = proto::S2Meta::new();
-    s2_meta.set_cells(::protobuf::RepeatedField::<proto::S2Cell>::from_vec(cells));
-    let attributes_meta = attributes
-        .iter()
-        .map(|(name, attribute)| {
-            let mut attr_meta = proto::Attribute::new();
-            attr_meta.set_name(name.to_string());
-            attr_meta.set_data_type(attribute.data_type().to_proto());
-            attr_meta
-        })
-        .collect();
-    s2_meta.set_attributes(::protobuf::RepeatedField::<proto::Attribute>::from_vec(
-        attributes_meta,
-    ));
-    meta.set_s2(s2_meta);
-    meta
-}
+    fn check_attributes(&mut self, batch: &PointsBatch) -> Result<()> {
```

Comment:
nit: maybe it is worth to comment this function, it inits with the first batch the attributes and then it checks for the following that the attributes are the same. this will not work in case of sparse attributes in the future

Reply:
Good idea, added a comment.